### ISSUE TITLE
Upgrade to jOOQ 3.17.9

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-boot#34668]

**Comment by esfomeado** [comment]
Shouldn't the upgrade be to 3.18.0 since Spring Boot ships `R2DBC` 1.0+ and jOOQ 3.17.x isn't compatible with it?

**Comment by wilkinsona** [comment]
No. Please see our [dependency upgrade policy](https://github.com/spring-projects/spring-boot/wiki/Team-Practices#third-party-dependency-upgrades) for details. An [upgrade to 3.18.0](https://github.com/spring-projects/spring-boot/issues/34576) has already been applied in 3.1.
